cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultApplicationContext.java

```diff
@@ -26,8 +32,30 @@
  */
 public class DefaultApplicationContext<T extends Config> implements ApplicationContext<T> {
 
+  private static final Gson GSON = new Gson();
+  private final T appConfig;
+
+  public DefaultApplicationContext() {
+    this.appConfig = null;
+  }
+
+  //We know that the type of the configToken instance will be T.
+  @SuppressWarnings("unchecked")
+  public DefaultApplicationContext(Class<? extends Application> app, @Nullable String configString)
```

Comment:
hm sorry we've visited this multiple times, but I think it would be cleaner to just pass in the config and do the deserialization outside of this class.